### ISSUE TITLE
average_precision_score: the ranking metric the family is visibly missing, and it is a sum rather than a trapezoid

### ISSUE THREAD

**Issue by CyrilB1531** [CyrilB1531/lodestar#210]
Measured against scikit-learn 1.9.0: `sklearn.metrics` exposes 61 metric
functions and `docs/equivalence.md` maps 33 of them. `average_precision_score`
is the most used of the ones it does not, and it is the one that leaves the
family #173 and #201 build visibly incomplete — a reader who reaches for a
ranking metric reaches for this one about as often as for NDCG.

```
average_precision_score(y_true, y_score, *, average='macro', pos_label=1, sample_weight=None)
```

Measured: `0.8333333333333333` on the binary case `y_true = [0, 0, 1, 1]`,
`y_score = [0.1, 0.4, 0.35, 0.8]`, and `0.3333333333333333` on the 2×3
multilabel case #201's corpus already uses.

## Why it is not simply "the area under the precision-recall curve"

It is a **sum, not an integral**: `Σ (Rₙ − Rₙ₋₁) · Pₙ` over the thresholds, which
is deliberately *not* the trapezoidal rule `auc(recall, precision)` applies.
scikit-learn's own documentation says so, because the trapezoid is optimistic on
this curve. Reproducing the wrong one is the mistake this issue exists to avoid,
and the two differ on the fixtures above by enough for a corpus to catch.

That also means it does **not** need `precision_recall_curve` to be public
first: the sum is over the same sorted thresholds `RocAuc` already walks in
`Internal/BinaryRoc.cs`. Whether it reuses that walk or gets its own is the
design question.

## What has to be decided rather than transcribed

- **Binary and multilabel, or binary alone.** `average='macro'` over a label
  matrix is the multilabel form, which shares its input shape with #201's three
  metrics. Shipping only the binary form is defensible; shipping the multilabel
  one afterwards is a second lot, and it should be a decision rather than a
  drift.
- **`pos_label`**, which `RocAuc.Score` already makes explicit where
  scikit-learn infers it — the same treatment applies here.
- **The undefined case.** With no positive sample scikit-learn warns
  ("No positive class found in y_true, recall is set to one for all thresholds")
  and returns a value; this package turns warnings into `ZeroDivision` where it
  can ([`decisions/0020`](docs/decisions/0020-normalize-is-a-projection-not-a-parameter.md)),
  and whether that applies here is a measured question.

## Evidence

A frozen corpus at `1e-9` including a fixture where the trapezoid and the sum
disagree, an equivalence row in the same commit, and reference pages under
`docs/reference/metrics/ranking/` that the gate enforces.
